src/Compilers/CSharp/Portable/Operations/CSharpOperationFactory.cs

```diff
@@ -304,8 +304,13 @@ private IInvocationExpression CreateBoundCallOperation(BoundCall boundCall)
             return new LazyInvocationExpression(targetMethod, instance, isVirtual, argumentsInEvaluationOrder, _semanticModel, syntax, type, constantValue, isImplicit);
         }
 
-        private ILocalReferenceExpression CreateBoundLocalOperation(BoundLocal boundLocal)
+        private IOperation CreateBoundLocalOperation(BoundLocal boundLocal)
         {
+            if (boundLocal.Syntax.Kind() == SyntaxKind.CatchDeclaration)
```

Comment:
>if (boundLocal.Syntax.Kind() == SyntaxKind.CatchDeclaration) [](start = 12, length = 60)

I think it would be better to have this special case in code that deals with BoundCatchBlock rather than making this function context aware.